internal/idutils/external_id.go

```diff
@@ -19,25 +19,31 @@ package idutils
 import (
 	"encoding/base64"
 	"fmt"
+	"github.com/dynatrace/dynatrace-configuration-as-code/pkg/config/v2/coordinate"
+	"strings"
 )
 
-// GenerateExternalID generates the externalID for settings 2.0 objects based on the schema, and ID.
-// The result of the function is pure.
-// Max length for the external ID is 500
-func GenerateExternalID(schema, id string) string {
+// GenerateExternalID returns a string that can be used as an external ID for a Settings 2.0 object
+func GenerateExternalID(coordinate coordinate.Coordinate) (string, error) {
+	if coordinate.Type == "" || coordinate.ConfigId == "" {
+		return "", fmt.Errorf("schema id and config id needs to be set to generate an external id for a settings 2.0 object")
+	}
+	if coordinate.Project == "" {
+		return generateExternalID(coordinate.Type, coordinate.ConfigId), nil
+	}
+	return generateExternalID(coordinate.Project, coordinate.Type, coordinate.ConfigId), nil
+}
+
+func generateExternalID(parts ...string) string {
 	const prefix = "monaco:"
-	const format = "%s$%s"
 	const externalIDMaxLength = 500
 
-	formattedID := fmt.Sprintf(format, schema, id)
-	encodedID := base64.StdEncoding.EncodeToString([]byte(formattedID))
-
-	encodedIDMaxLength := externalIDMaxLength - len(prefix)
-	if len(encodedID) > encodedIDMaxLength {
-		encodedID = encodedID[encodedIDMaxLength:]
+	rawID := strings.Join(parts, "$")
+	encodedID := base64.StdEncoding.EncodeToString([]byte(rawID))
+	if len(encodedID) > externalIDMaxLength-len(prefix) {
+		encodedID = encodedID[len(encodedID)-(externalIDMaxLength-len(prefix)):]
```

Comment:
hmm... you changed how a >500 character externalID is cut off. Won't that mean that we can't detect long old externalIDs anymore? 

So changing this will impact/break the extID update in this case